include/blas_quda.h

```diff
@@ -408,40 +478,58 @@ namespace quda {
        @param[in] x input vector
        @param[in,out] y update vector
     */
-    double2 axpyCGNorm(double a, const ColorSpinorField &x, ColorSpinorField &y);
+    cvector<double2> axpyCGNorm(cvector<double> &a, cvector_ref<const ColorSpinorField> &x,
+                                cvector_ref<ColorSpinorField> &y);
 
     /**
        @brief Computes ||x||^2, ||r||^2 and the MILC/FNAL heavy quark
        residual norm
        @param[in] x input vector
        @param[in] r input vector (residual vector)
     */
-    double3 HeavyQuarkResidualNorm(const ColorSpinorField &x, const ColorSpinorField &r);
+    cvector<double3> HeavyQuarkResidualNorm(cvector_ref<const ColorSpinorField> &x,
+                                            cvector_ref<const ColorSpinorField> &r);
+
+    inline double3 HeavyQuarkResidualNorm(const ColorSpinorField &x, const ColorSpinorField &r)
```

Comment:
Similar comment as above, any reason why we need a separate non-`cvector_ref` version here?

Reply:
See comment above.